src/test/tree/EigenPodManager.tree

```diff
@@ -0,0 +1,95 @@
+├── EigenPodManager Tree (*** denotes that integrationt tests are needed to validate path)
+├── when contract is deployed and initialized
+│   └── it should properly set storage
+├── when initialize called again
+│   └── it should revert
+├── when createPod called
+│   ├── given the user has already created a pod
+│   │   └── it should revert
+│   ├── given that the max number of pods has been deployed
+│   │   └── it should revert
+│   └── given the user has not created a pod
+│       └── it should deploy a pod
+├── when stake is called
+│   ├── given the user has not created a pod
+│   │   └── it should deploy a pod
+│   └── given the user has already created a pod
+│       └── it should call stake on the eigenPod
+├── when setMaxPods is called
+│   ├── given the user is not the pauser
+│   │   └── it should revert
+│   └── given the user is the pauser
+│       └── it should set the max pods
+├── when updateBeaconChainOracle is called
+│   ├── given the user is not the owner
+│   │   └── it should revert
+│   └── given the user is the owner
+│       └── it should set the beacon chain oracle
+├── when addShares is called
+│   ├── given that the caller is not the delegationManager
+│   │   └── it should revert
+│   ├── given that the podOwne address is 0
+│   │   └── it should revert
+│   ├── given that the shares amount is negative
+│   │   └── it should revert
+│   ├── given that the shares is not a whole gwei amount
+│   │   └── it should revert
+│   ├── given that too many shares are withdrawn
+│   │   └── it should revert
+│   └── given that all of the above conditions are satisfied
+│       └── it should update the podOwnerShares
+├── when removeShares is called
+│   ├── given that the caller is not the delegationManager
+│   │   └── it should revert
+│   ├── given that the shares amount is negative
+│   │   └── it should revert
+│   ├── given that the shares is not a whole gwei amount
+│   │   └── it should revert
+│   ├── given that too many shares are removed
+│   │   └── it should revert
```

Comment:
I'm unsure what the correct amount of detail is for these tree diagrams, but think that "too many shares are removed" is quite vague -- specifically, we revert if the amount removed would cause the podOwner's shares to become negative.
Let me know if you think being this specific in the tree diagram is overkill! I'm not personally sure what's better, since more concise is more readable.

Reply:
We should strive to be as concise as possible to get the gist of the execution. Updated this to be: `given that removing shares results in the pod owner having negative shares`

Reply:
cool, works for me, ty ty